kaldb/src/main/java/com/slack/kaldb/elasticsearchApi/OpenSearchRequest.java

```diff
@@ -37,9 +49,62 @@
  * building a complete working list of queries to be performed.
  */
 public class OpenSearchRequest {
+  public static final Logger LOG = LoggerFactory.getLogger(OpenSearchRequest.class);
+
   private static final ObjectMapper OM =
       new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
 
+  // key - index. value - list of docs to be indexed
+  public static Map<String, List<Trace.Span>> convertIndexRequestToTraceFormat(
+      List<IndexRequest> indexRequests) {
+    Map<String, List<Trace.Span>> indexDocs = new HashMap<>();
+
+    for (IndexRequest indexRequest : indexRequests) {
+      String index = indexRequest.index();
+      IngestDocument ingestDocument = convertRequestToDocument(indexRequest);
+      List<Trace.Span> docs = indexDocs.computeIfAbsent(index, key -> new ArrayList<>());
+      docs.add(MurronLogFormatter.fromIngestDocument(ingestDocument));
+    }
+    return indexDocs;
+  }
+
+  // only parse IndexRequests
+  public static List<IndexRequest> parseBulkRequest(String postBody) throws IOException {
+    List<IndexRequest> indexRequests = new ArrayList<>();
+    BulkRequest bulkRequest = new BulkRequest();
+    // calls parse under the hood
+    bulkRequest.add(
+        postBody.getBytes(StandardCharsets.UTF_8), 0, postBody.length(), null, XContentType.JSON);
+    List<DocWriteRequest<?>> requests = bulkRequest.requests();
+    for (DocWriteRequest<?> request : requests) {
+      if (request.opType() == DocWriteRequest.OpType.INDEX) {
+
+        // The client makes a DocWriteRequest and sends it to the server
+        // IngestService#innerExecute is where the server eventually reads when request is an
+        // IndexRequest. It then creates an IngestDocument
+        indexRequests.add((IndexRequest) request);
+      } else {
+        LOG.warn(
+            "request=" + request + " of type " + request.opType().toString() + "not supported");
+      }
+    }
+    return indexRequests;
+  }
+
+  protected static IngestDocument convertRequestToDocument(IndexRequest indexRequest) {
+    String index = indexRequest.index();
```

Comment:
Is there a reason we create all of these as separate variables?  Did we consider a static method on IngestDocument that has a `fromIndexRequest()`?

Reply:
Disregard - I missed the fact that `IngestDocument` wasn't our class.